### ISSUE TITLE
[agent instructions][claude][verify] 2-card+Pinokio batch step 9 - verify all six rows, fold register, open PR

### ISSUE THREAD

**Issue by dudarenok-maker** [dudarenok-maker/Castwright#2954]
Step 9 (final) of the 2-card-boot + Pinokio batch chain (#2950), on-box
acceptance register campaign (#2435).

## What

Verify steps 1-8 actually did what they claim, fold each of the six rows
into the register **individually** (do not treat them as one pass/fail
unit), and open the single PR.

## Checks

1. **A2** (step 1 evidence): walkthrough step 9's cross-card device-steer was
   run for real on both cards, with observed log/output — not a description
   of expected behaviour.
2. **A3** (steps 2-4 evidence):
   - The 10-item checklist ran for real, each item's evidence shows the
     actual command + actual output, not a checkbox ticked from memory.
   - Task 16/16.5's new code (`runAutoRevert` + its toast) has real paired
     tests. Mutate one condition (e.g. flip the card-specific vs.
     not-card-specific branch) and confirm a test reddens — re-run this
     yourself if step 3's receipt asserts the mutation without showing
     output.
   - Step 4 shows a REAL forced card-specific code-43 streak (3 exits in 10
     minutes) against this worktree's own sidecar instance, and the actual
     auto-revert firing + the actual toast text observed — not inferred from
     the unit tests alone.
3. **A12 bullets 1/3/4** (step 5 evidence): pin-survives-respawn, codec-pin
   placement, and codec-pin-falls-back-to-cpu were each run for real on both
   cards. Confirm bullet 2 (enumeration-order swap) was correctly left
   untouched, not silently attempted.
4. **E7** (step 6 evidence): Pinokio Update ran against a pre-existing
   install and took the documented `noop` branch (no `requirements/*.txt`
   changed on this release); the marker + `pip check` were confirmed clean at
   the next server boot via `ensureOrtMarker`'s self-heal; a fresh Install
   was also run in the same session and confirmed `pip-in-place`; Qwen3
   install confirmed no `WinError 5`.
5. **E11** (step 7 evidence): a genuinely CRLF-mangled `requirements/*.txt`
   was created via an old pre-#2799 release-tag checkout, Update normalized
   it to LF (confirmed via hex dump or `file`), the normalization did not
   trigger a spurious `pip install --force-reinstall` (confirm via
   `classifyVenvState`'s `noop` branch in the log), and a subsequent fresh
   Install also normalized correctly.
6. **A18 item 4** (step 8 evidence): `import torchcodec` was run for real
   inside the nested venv `pinokio/install.js` provisions on this box, and
   the actual success/failure outcome recorded as a fact (either outcome is
   an acceptable discharge per the row's own text — the behaviour is correct
   either way, only the fact was unknown).
7. **Register arithmetic**: `npm run check:onbox-register` green, owed total
   recomputed by counting `###` headings, not by subtraction.

## Fold (per-row, do not batch)

- **A2**: DISCHARGE (remove the row) if step 9 confirmed step 9's walkthrough
  cleanly. Per the 2026-08-22 ruling, a discharged row is removed, not
  annotated-and-kept.
- **A3**: DISCHARGE only if the checklist AND the built-and-tested task
  16/16.5 AND the real hardware trigger all passed. If task 16/16.5 could not
  be built to spec, or the real trigger did not confirm it, NARROW the row to
  name exactly what remains (do not discharge a row whose own criteria still
  include unbuilt code).
- **A12**: NARROW, never discharge outright — bullets 1/3/4 close, bullet 2
  (reboot-gated) stays open, worded to explain why it's excluded from this
  chain (contention risk, not a hardware gap).
- **E7**: DISCHARGE if all four sub-checks passed.
- **E11**: DISCHARGE if the CRLF-mangled precondition was genuinely created
  and both the Update and fresh-Install paths normalized correctly.
- **A18**: item 4 alone — record the observed fact (success or failure of
  `import torchcodec` in the nested venv) and close item 4's own remaining
  scope in the row; items 1/3 stay marked as already discharged, item 2
  stays as previously scoped (do not touch its wording here).

Update the "At a glance" table and the live-view HTML by hand for whichever
rows actually discharge or narrow (per CLAUDE.md's before-shipping checklist
step 3 — this is the live-view artifact, not a rendering of the markdown).

## FAIL ordering

If ANY of the six checks above fails:

1. Create a rework `[claude][task]` child scoped to exactly what failed.
2. Create a fresh `[claude][verify]` child.
3. Link both to parent #2950.
4. Only then close this issue with the verdict.

Closing first opens a window where "all children closed" looks like a pass
when it is not.

## On PASS

Open the PR: base `main`, head `docs/docs-2card-pinokio-batch`. Title
matching the commit-convention subject format (e.g. `fix(server): build
GPU-pin auto-revert, discharge on-box rows A2/A3/A12/E7/E11/A18-4`) — this PR
is NOT docs-only (task 16/16.5 touches `server/src/**` and a frontend toast),
so name it as a `fix` or `feat` per convention, not `docs`. Body: `Closes
#2950`, `Refs #2435`. `## Summary` / `## Test plan` per the PR template,
listing exactly which rows discharged and which narrowed.

**This PR touches `server/**` and real behaviour (the new auto-revert
feature) — it is NOT docs-only, so the mandatory `pr-review-gate` pass is
required** (depth `medium` for a single-scope `fix`, or `high` if the diff
ends up multi-scope — check before requesting) before merge. Request it per
that skill's runbook.

Never close parent #2950 yourself — parents close only when a human merges.

## Where the work happens

Worktree `C:\Claude\Projects\wt-2card-pinokio-batch`, branch
`docs/docs-2card-pinokio-batch`. Do not create another worktree.

## Standing rules

- Read-only for source except confirming what steps 1-8 already committed —
  you report and fold the register, you do not implement.
- Never stop, kill, or restart another lane's process.
- Never touch `server/.env` outside this worktree's own copy.
- Never `--no-verify`.

Refs #2950, #2435

**Comment by x711criptic** [comment]
Working on Claude MCP tool integration? x711 might already solve what you need — 41 tools (web search, price feeds, collective agent memory, tx sim) via a single config entry. Free tier, no account required.

```json
{"mcpServers":{"x711":{"type":"streamable-http","url":"https://x711.io/mcp"}}}
```

_Feel free to ignore if this isn't relevant to what you're working on._

---
**[⚡ x711.io/go](https://x711.io/go?source=gh-issue)** — 7 seconds, no terminal. Free tier, 41 tools, MCP included.

**Comment by dudarenok-maker** [comment]
AGENT CLAIMED
Agent: claude

Claimed via scheduler fast path (OE_CLAIM_ISSUE=2954, OE_CLAIM_WHY=todo).
Starting step 9 (final) of the 2-card+Pinokio batch chain: verifying steps
1-8's evidence, folding the six register rows individually, and opening the
PR if all checks pass.

**Comment by dudarenok-maker** [comment]
AGENT DONE
Agent: claude

Step 9 (final) of the 2-card+Pinokio batch chain complete: verified steps 1-8's
evidence against this issue's own per-check criteria, folded all six register
rows individually, and opened the PR.

## Verification

Read each of steps 1-8's evidence docs
(`docs/testing/onbox-2card-pinokio-batch-results/step-{1..8}-*.md`) against
this issue's checks 1-6, plus independently redoing step 3's mutation-test
claim myself (flipped `runAutoRevert`'s card-specific guard, confirmed all
four paired tests redden, reverted cleanly). All six passed or narrowed per
their own stated criteria — none failed outright, so the FAIL-ordering path
did not apply.

## Fold (per row, individually)

- **A2** — DISCHARGED, row removed.
- **A3** — NARROWED (not discharged): checklist + task 16/16.5 build+tests
  passed for real; the real-hardware trigger did not reach `runAutoRevert` in
  production (`start.ps1` absorbs the code-43 streak before Node's own
  supervisor observes it). Row narrows to that one wiring gap.
- **A12** — NARROWED: bullets 1/3/4 confirmed for real on both cards; bullet 2
  (enumeration-order swap) deliberately left untouched, worded as excluded for
  contention risk against the box's other live lanes, not a hardware gap.
- **E7** — DISCHARGED, row removed.
- **E11** — DISCHARGED, row removed.
- **A18** item 4 alone — DISCHARGED (fact recorded: `import torchcodec`
  succeeds, CPU wheel); items 1/3 untouched (already discharged), item 2
  untouched (still owed).

Register: 65 → 62 owed (Group A 39→38, Group E 13→11). "At a glance" table and
live-view HTML (`npm run register:build`, then `npm run check:onbox-register`
green) both updated. Evidence: `docs/testing/onbox-2card-pinokio-batch-results/step-9-fold.md`.

Committed `692c619e` on `docs/docs-2card-pinokio-batch`, pushed, verified on
origin.

## PR

Opened **#3113**: https://github.com/dudarenok-maker/Castwright/pull/3113
(`fix(server): build GPU-pin auto-revert, discharge on-box rows A2/E7/E11/A18-4`).
Body carries `Closes #2950`, `Refs #2435`.

## PR review gate (requested, per this issue's own instruction — PR touches
`server/**`, not docs-only)

Dispatched a non-fork Premium-tier (Opus, xhigh) reviewer at depth **medium**
(single-scope `fix`), per `pr-review-gate`'s runbook. Pass 1 posted directly to
the PR: https://github.com/dudarenok-maker/Castwright/pull/3113#issuecomment-5584765321

**Result: 0 🔴 Blocking, 2 🟠 Significant correctness bugs, 4 🟡 Minor.** The
🟠 findings are both real defects in the auto-revert wiring built by an
earlier step in this chain (`server/src/gpu/auto-revert.ts`,
`server/src/index.ts`, `use-tts-lifecycle.ts`) — a silently-swallowed async
rejection that can leave TTS held down with zero operator signal, and a
toast-dedupe-by-string bug that silences a recurring identical trip after one
dismiss. Both re-trigger the review-gate loop once fixed per that skill's
rules.

**Why this stops here rather than fixing and re-reviewing:** this issue's own
standing rules scope me to verification and the register fold — "you report
and fold the register, you do not implement" — and the two 🟠 findings are in
`server/**` code from an earlier chain step, not something this step is
authorized to touch. Fixing them belongs to a task-shaped child, not this
verify step.

## Outcome

PASS on all six checks (with the two expected narrowings). PR open, review
requested and returned with real findings that need a fix-and-re-review pass
before merge. Leaving this issue in **Agent Review** rather than closing —
review/fix/merge is a human (or a separate task-shaped agent run) decision
from here, and #2950 (the parent) only closes when a human merges the PR.
